client/src/pages/Balance/ReactStripe.js

```diff
@@ -0,0 +1,304 @@
+    event.preventDefault();
+    if (!stripe || !elements) {
+      // Stripe.js has not loaded yet. Make sure to disable
+      // form submission until Stripe.js has loaded.
+      return;
+    }
+
+    if (error) {
+      elements.getElement("card").focus();
+      return;
+    }
+
+    if (cardComplete) {
+      setProcessing(true);
+    }
+
+    const requestPaymentIntent = await axios.post(
+      `/user/${user._id}/purchase-credit`,
+      {
+        credits
+      }
+    );
+
+    const payload = await stripe.createPaymentMethod({
+      type: "card",
+      card: elements.getElement(CardElement),
+      billing_details: billingDetails
+    });
+
+    const confirm = await stripe.confirmCardPayment(
+      `${requestPaymentIntent.data.clientSecret}`,
+      {
+        payment_method: payload.paymentMethod.id
+      }
+    );
+
+    if (confirm.paymentIntent.status === "succeeded") {
+      setMadeSuccessfulPayment(true);
+      setProcessing(false);
+      const { data } = await axios.put(`/user/${user._id}/add-credit`, {
+        credits
+      });
+      if (data.success) {
+        setMadeSuccessfulPayment(true);
+        user.credits += credits;
+      }
+    }
+  };
+
+  const goToAddCredits = () => {
+    setMadeSuccessfulPayment(false);
+    setAddCredit(true);
+  };
+
+  const reset = () => {
+    setError(null);
+    setProcessing(false);
+    setPaymentMethod(null);
+    setBillingDetails({ name: "" });
+  };
+
+  return madeSuccessfulPayment ? (
+    <>
+      <Grid container direction="column" spacing={6}>
+        <Grid item xs={12}>
+          <Typography className={classes.text}>Payment Complete</Typography>
+        </Grid>
+        <Grid container item direction="row" xs={12} justify="center">
+          <Grid item xs={6}>
+            <Button
+              className={classes.button}
+              color="primary"
+              variant="contained"
+              onClick={goToAddCredits}
+            >
+              View Balance
+            </Button>
+          </Grid>
+          <Grid item xs={6}>
+            <Link to="/code-upload" className={classes.link}>
+              <Button
+                className={classes.button}
+                color="primary"
+                variant="contained"
+              >
+                Upload Code
+              </Button>
+            </Link>
+          </Grid>
+        </Grid>
+      </Grid>
+    </>
+  ) : (
+    <form className="Form" onSubmit={handleSubmit}>
+      <fieldset className="FormGroup">
+        <Field
+          label="Name"
+          id="name"
+          type="text"
+          placeholder="Jane Doe"
+          required
+          autoComplete="name"
+          value={billingDetails.name}
+          onChange={e => {
+            setBillingDetails({ ...billingDetails, name: e.target.value });
+          }}
+        />
+      </fieldset>
+      <fieldset className="FormGroup">
+        <CardField
+          onChange={e => {
```

Comment:
This CardField and the name Field above it are display at different heights. Should probably be the same. Not a big deal, but it was perceptible from afar.

Reply:
I am not seeing this issue on my machine. Not sure what you mean.